superset-frontend/packages/superset-ui-core/src/connection/callApi/parseResponse.ts

```diff
@@ -30,7 +36,8 @@ export default async function parseResponse<T extends ParseMethod = 'json'>(
     : T extends 'text'
     ? TextResponse
     : never;
-  const response = await apiPromise;
+
+  const response = isPromise(apiRequest) ? await apiRequest : apiRequest;
```

Comment:
```suggestion
  const response = await Promise.resolve(apiRequest);
```
You don't really need the `isPromise` check